pkg/apis/validate/metadata.go

```diff
@@ -41,5 +43,15 @@ func ObjectMetadata(meta metav1.Object) *apis.FieldError {
 			Paths:   []string{"name"},
 		}
 	}
+
+	// metadata must not contain any label with tekton.dev/*
+	for k := range meta.GetLabels() {
+		if strings.HasPrefix(k, pipeline.GroupName) {
+			return &apis.FieldError{
+				Message: "Invalid label key: can not change tekton resource name",
```

Comment:
NIT: the error message may be confusing if someone is not trying to change the resource name. I think it would be better to say that the `tekton.dev/` labels are reserved for internal use.